src/Microsoft.DotNet.Darc/Darc/Operations/UpdateSubscriptionOperation.cs

```diff
@@ -114,6 +124,84 @@ public override async Task<int> ExecuteAsync()
             {
                 excludedAssets = [.._options.ExcludedAssets.Split(';', StringSplitOptions.RemoveEmptyEntries)];
             }
+
+            if (_options.OverwriteMergePolicies)
+            {
+                mergePolicies = [];
+            }
+
+            // Parse the merge policies
+            if (_options.AllChecksSuccessfulMergePolicy)
+            {
+                AddMergePolicyWithIgnoreChecksIfMissing(mergePolicies, MergePolicyConstants.AllCheckSuccessfulMergePolicyName);
+            }
+
+            if (_options.NoRequestedChangesMergePolicy && !mergePolicies.Any(p => p.Name == MergePolicyConstants.NoRequestedChangesMergePolicyName))
+            {
+                mergePolicies.Add(
+                    new MergePolicy
+                    {
+                        Name = MergePolicyConstants.NoRequestedChangesMergePolicyName,
+                        Properties = []
+                    });
+            }
+
+            if (_options.DontAutomergeDowngradesMergePolicy && !mergePolicies.Any(p => p.Name == MergePolicyConstants.DontAutomergeDowngradesPolicyName))
+            {
+                mergePolicies.Add(
+                    new MergePolicy
+                    {
+                        Name = MergePolicyConstants.DontAutomergeDowngradesPolicyName,
+                        Properties = []
+                    });
+            }
+
+            if (_options.StandardAutoMergePolicies)
+            {
+                AddMergePolicyWithIgnoreChecksIfMissing(mergePolicies, MergePolicyConstants.StandardMergePolicyName);
+            }
+
+            if (_options.ValidateCoherencyCheckMergePolicy && !mergePolicies.Any(p => p.Name == MergePolicyConstants.ValidateCoherencyMergePolicyName))
+            {
+                mergePolicies.Add(
+                    new MergePolicy
+                    {
+                        Name = MergePolicyConstants.ValidateCoherencyMergePolicyName,
+                        Properties = []
+                    });
+            }
+
+            if (_options.SourceFlowCheckMergePolicy)
```

Comment:
This snippet is quite similar to the one in `AddSubscriptionOperation` and seems to be tied to source flow rather than the operations themselves. Should it be maybe in some helper object instead?